conbench/app/_plots.py

```diff
@@ -186,41 +187,67 @@ def _source(
     # the tooltip?
     commit_messages = [d["message"] for d in data]
 
-    dates = [dateutil.parser.isoparse(x["timestamp"]) for x in data]
+    # Note(JP): isoparse() returns a `datetime.datetime` object. And I think
+    # that the `timestamp` property corresponds to the utc-local invocation (or
+    # finish) time of the corresponding benchmark case run (the `timestamp`
```

Comment:
The timestamp property here comes from `/api/history`, which means it's actually the commit timestamp. That comes straight from GitHub and represents a UTC time. However, the `/api/history` endpoint just reports it as e.g. `"2022-03-03T14:16:51"` so this datetime will be tz-naive, assumed to be in UTC.

Reply:
Big thanks for this. I forgot to ping you on this specifically. Will update the comment and also the relevant pieces of text here.

> that comes straight from GitHub and represents a UTC time

Anecdote about this one (I think I have shared this before but :shrug:): the GitHub HTTP API shows tz-aware Zulu timestamps like `2011-04-14T16:00:49Z`. That is, those timestamps _unambiguously_ refer to UTC timezone. And then pygithub came in and in my opinion did a big mistake to transparently interpret these timestamps ans tz-naive ones. See `https://github.com/PyGithub/PyGithub/issues/512#issuecomment-1362654366.

When  we store and read commit timestamps -- let's make sure we treat the timezone information explicitly so that after all when data is displayed to users that there is no timezone ambiguity. I quickly created a reminder: https://github.com/conbench/conbench/issues/609.